### ISSUE TITLE
Admission Controller Webhook for complex object validation

### ISSUE THREAD

**Issue by itamar-marom** [datainfrahq/druid-operator#30]
Object validation should not be done inside the `Reconcile` function. As some logic moved info Kubebuilder markers, some object validation is still left in the `verifyDruidSpec` function.</br>
Validations should be done via [Admission Controller Webhook](https://book.kubebuilder.io/reference/webhook-overview.html).</br>

If we agree, I'll take that as soon as the Kubebuilder v3 migration will end.

**Comment by AdheipSingh** [comment]
yes agreed. The ```verifyDruidSpec``` has been there since the start, at the point operator-sdk dint even have any admission webhook in its cli not openAPI validation, so we ended up not implementing this.

Regardless, isnt open API sufficient for the validations of the CR. What are you plans to implement in acw ?

**Comment by itamar-marom** [comment]
First I'll try to move the regex also into marker validation.  
The problems come with the `drd.Spec.Image == "" && node.Image == ""` condition which I don't think Kubebuilder markers have the capability for it - In that case we will need a validatingHook that will hold this logic.

**Comment by itamar-marom** [comment]
Also, I don't think that Kubebuilder supports the validation of a key on a map object. So this will probably will sit in the webhook also: Nodes map[string]DruidNodeSpec `json:"nodes"`

**Comment by itamar-marom** [comment]
@AdheipSingh Looking at [Kubebuilder webhook documentation](https://book.kubebuilder.io/cronjob-tutorial/cert-manager.html) it looks like cert-manager is mandatory for the API server - webhook communication. How should we handle that in your opinion? 
You can look at the MR, I created a webhook and put the validation logic inside.
